pkg/storage/etcd3/store.go

```diff
@@ -448,13 +448,6 @@ func (s *store) ttlOpts(ctx context.Context, ttl int64) ([]clientv3.OpOption, er
 	return []clientv3.OpOption{clientv3.WithLease(clientv3.LeaseID(lcr.ID))}, nil
 }
 
-func keyWithPrefix(prefix, key string) string {
-	if strings.HasPrefix(key, prefix) {
-		return key
```

Comment:
@lavalamp

If you don't think there is a key "that didn't get the prefix prepended", then we are actually correcting this behavior.